src/routes/events.rs

```diff
@@ -0,0 +1,31 @@
+use actix_web_lab::sse::{Data, SendError, Sender};
+
+use crate::sse_events;
+
+pub async fn emit<T: Into<Data>>(sender: &Sender, event: T) -> Result<(), SendError> {
+    sender.send(event.into()).await?;
+    //Empty message to force send the above message to receiver
+    //Else, will stay in the buffer when using actix_rt::spawn
+    //TODO: Investigate further to avoid this workaround
+    sender.send(Data::new("")).await?;
```

Comment:
Interesting.

It doesn't look like there's a `flush` method or anything. Odd that the first `send` isn't getting through to the server? Without the second empty send, does it never make it to the server?

Reply:
I couldn't find a way to check for the buffered content. But the message doesn't get to the client without the second send. This problem seems to only arise when calling this from within `actix_rt::spawn`.